eva_cttv_pipeline/clinvar_xml_utils.py

```diff
@@ -435,3 +438,64 @@ def sequence_location_helper(self, attr):
             # TODO: https://github.com/EBIvariation/eva-opentargets/issues/172
             return None
         return sequence_locations[0].attrib.get(attr)
+
+    def get_variant_name_or_hgvs(self):
+        if self.preferred_or_other_name:
+            return self.preferred_or_other_name
+        if self.toplevel_refseq_hgvs:
+            return self.toplevel_refseq_hgvs
+        return None
+
+    def _parse_identifier(self):
+        if self._parsed_identifier is None:
+            self._parsed_identifier = clinvar_identifier_parsing.parse_variant_identifier(self.get_variant_name_or_hgvs())
+        return self._parsed_identifier
+
+    @property
+    def transcript_id(self):
+        return self._parse_identifier()[0]
+
+    @property
+    def coordinate_span(self):
+        parsed_coordinate_span = self._parse_identifier()[1]
+        return parsed_coordinate_span if parsed_coordinate_span is not None else self.explicit_insertion_length
+
+    @property
+    def repeat_unit_length(self):
+        return self._parse_identifier()[2]
+
+    @property
+    def is_protein_hgvs(self):
+        return self._parse_identifier()[3]
```

Comment:
These properties will be available to any `ClinVarRecordMeasure` right? I'm wondering if their names should reflect that they are only relevant in the context of a repeat or that they come from parsing the hgvs

Reply:
I have extracted the properties to a new class and added `hgvs_properties` to `ClinVarRecordMeasure` for accessing them. Parsing the HGVS name will happen the first time you access `hgvs_properties`.